packages/visualizations/src/components/Map/MapRender.svelte

```diff
@@ -1,73 +1,85 @@
 <svelte:options immutable={true} />
 
-<script>
-    /*
-TODO:
-- Restrict movement:
-    - Block minZoom to the initial one after fitting the bbox
-    - Compute maxZoom based on the smallest extent when using queryFeature restricted to this bbox
-- Display the map based on style, source, layer, refresh when one changes
-- Adapt display based on the size of the map (single, main, side)
-*/
-    import maplibregl from 'maplibre-gl';
+<script lang="ts">
+    import maplibregl, {
+        Map,
+        SourceSpecification,
+        StyleSpecification,
+        NavigationControl,
+        LngLatBoundsLike,
+        MapSourceDataEvent,
+        MapMouseEvent,
+        LngLatLike,
+    } from 'maplibre-gl';
     import { onMount } from 'svelte';
     import { debounce } from 'lodash';
+    // eslint-disable-next-line import/no-unresolved
+    import type { BBox } from 'geojson';
     import { computeMaxZoomFromGeoJsonFeatures, getFixedTooltips } from './utils';
     import ColorsLegend from '../utils/ColorsLegend.svelte';
+    import type { ColorsScale, DataBounds, LegendVariant } from '../types';
+    import type {
+        ChoroplethFixedTooltipDescription,
+        MapLayer,
+        MapRenderTooltipFunction,
+        MapLegend,
+    } from './types';
 
     // maplibre style (basemap)
-    export let style;
+    export let style: StyleSpecification;
     // maplibre source config
-    export let source;
+    export let source: SourceSpecification;
     // maplibre layer config
-    export let layer;
+    export let layer: MapLayer;
     // bounding box to start from, and restrict to it
-    export let bbox;
-
+    export let bbox: BBox;
```

Comment:
I was a bit torn here initially, because `LngLatBoundsLike` comes from Maplibre, and I'd rather expose things that are generic. But overall I think I prefer to require a parameter in a specific non-ambiguous format instead of tolerating many and having to support and sometimes transform them, and the bbox one is the most straight-forward (`[number, number, number, number]`).